src/Conf.ml

```diff
@@ -1615,17 +1627,47 @@ let xdg_config =
   | Some xdg_config_home ->
       let filename =
         Fpath.(
-          append (v xdg_config_home) (v "ocamlformat" / ".ocamlformat"))
+          append (v xdg_config_home) (v "ocamlformat" / dot_ocamlformat))
       in
       if Fpath.exists filename then Some filename else None
   | None -> None
 
+let is_ignored ~quiet ~ignores ~filename =
+  let drop_line l = String.is_empty l || String.is_prefix l ~prefix:"#" in
+  (* process deeper files first *)
+  let ignores = List.rev ignores in
+  List.find_map ignores ~f:(fun (dir, ignore_file) ->
+      try
+        In_channel.with_file (Fpath.to_string ignore_file) ~f:(fun ch ->
+            let lines =
+              In_channel.input_lines ch
+              |> List.mapi ~f:(fun i s -> (i + 1, String.strip s))
+              |> List.filter ~f:(fun (_, l) -> not (drop_line l))
+            in
+            List.find_map lines ~f:(fun (lno, line) ->
+                match Fpath.of_string line with
+                | Ok file_on_current_line ->
+                    let f = Fpath.(dir // file_on_current_line) in
+                    if Fpath.equal filename f then Some (ignore_file, lno)
+                    else None
+                | Error (`Msg msg) ->
+                    if not quiet then
+                      Format.eprintf "File %a, line %d:\nWarning: %s\n"
+                        (Fpath.pp ~pretty:true) ignore_file lno msg ;
+                    None ) )
+      with Sys_error err ->
+        if not quiet then
+          Format.eprintf "Warning: ignoring %a, %s\n"
+            (Fpath.pp ~pretty:true) ignore_file err ;
+        None )
```

Comment:
In the case that ocamlformat is called with many files in one invocation, IIUC this rescans the ignore files repeatedly. I guess that this shouldn't be problematic. Since the ignore set depends on where each input file is, I guess it is nontrivial to memoize this.